src/transformers/models/patchtst/modeling_patchtst.py

```diff
@@ -0,0 +1,2091 @@
+                mask_ratio=self.mask_ratio,
+                unmasked_channel_indices=self.unmasked_channel_indices,
+                channel_consistent_masking=self.channel_consistent_masking,
+                mask_value=self.mask_value,
+                seed_number=self.seed_number,
+            )
+        elif self.mask_type == "forecast":
+            masked_input, mask = forecast_masking(
+                inputs=patch_input,
+                patch_lengths=self.mask_patches,
+                mix_ratio=self.mask_patch_ratios,
+                unmasked_channel_indices=self.unmasked_channel_indices,
+                mask_value=self.mask_value,
+                seed_number=self.seed_number,
+            )
+        else:
+            raise Exception("Invalid mask type")
+
+        mask = mask.bool()  # mask: [bs x num_input_channels x num_patch]
+
+        return masked_input, mask
+
+
+class PatchTSTEncoderBlock(nn.Module):
+    """
+    PatchTST encoder block
+    """
+
+    def __init__(self, config: PatchTSTConfig):
+        super().__init__()
+
+        self.layers = nn.ModuleList([PatchTSTEncoderLayer(config) for i in range(config.encoder_layers)])
+
+    def forward(self, hidden_state: torch.Tensor, output_hidden_states: bool = False):
+        """
+        Parameters:
+            hidden_state (`torch.Tensor` of shape `(batch_size, num_channels, sequence_length, d_model)`, *required*):
+                Past values of the time series
+            output_hidden_states (`bool`, *optional*, default to False):
+                output hidden state option
+        Return:
+            hidden_state (`torch.Tensor` of shape `(batch_size, num_channels, sequence_length, d_model)`)
+
+            all_hidden_states (*optional*, returned when `output_hidden_states` is set to True, tuple of `torch.Tensor`
+            of shapes `(batch_size, num_channels, sequence_length, d_model)`)
+
+        """
+        all_hidden_states = []
+
+        for mod in self.layers:
+            hidden_state = mod(hidden_state)
+            if output_hidden_states:
+                all_hidden_states.append(hidden_state)
+        if output_hidden_states is False:
+            return hidden_state, None
+        return hidden_state, all_hidden_states
+
+
+class PatchTSTEncoderLayer(nn.Module):
+    """
+    PatchTST encoder layer
+    """
+
+    def __init__(self, config: PatchTSTConfig):
+        super().__init__()
+
+        self.channel_attention = config.channel_attention
+
+        # Multi-Head attention
+        self.self_attn = PatchTSTAttention(
+            embed_dim=config.d_model,
+            num_heads=config.encoder_attention_heads,
+            dropout=config.attention_dropout,
+        )
+
+        # Add & Norm of the sublayer 1
+        self.dropout_path1 = nn.Dropout(config.dropout_path) if config.dropout_path > 0 else nn.Identity()
+        if "batch" in config.norm.lower():
+            self.norm_sublayer1 = nn.Sequential(
```

Comment:
we usually avoid the use of `nn.Sequential` as it's a pain to debug

Reply:
Thank you! nn.Sequential is removed.